synapse/push/push_rule_evaluator.py

```diff
@@ -232,6 +232,8 @@ def _flatten_dict(
     if result is None:
         result = {}
     for key, value in d.items():
+        if isinstance(value, bool):
+            value = str(value)
```

Comment:
I don't like this implementation because it will give `"True"` and `"False"` as strings. Other languages represent them differently. The encoding of booleans as strings should really be specified unambiguously so push rules can be written to work on any homeserver.

Reply:
Agreed, the below string check lowercases them but it's not explicit in any way.

Reply:
ah, well, it feels less bad then (Python's capitalisation is particularly unusual, I guess), but I suppose we still ought to make it explicit in the spec how things should be stringified if this is what we're doing.